app/mailers/oral_history_delivery_mailer.rb

```diff
@@ -0,0 +1,70 @@
+class OralHistoryDeliveryMailer < ApplicationMailer
+
+  def oral_history_delivery_email
+    raise ArgumentError.new("Required params[:request] missing") unless request.present?
+    raise ArgumentError.new("Required patron email missing") unless request.patron_email.present?
+    mail(to: to_address, subject: subject, content_type: "text/html")
+  end
+
+  def request
+    @request ||= params[:request]
+  end
+
+  def to_address
+    request.patron_email
+  end
+
+  # Note: any value greater than 604800 will raise an exception.
+  # See https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html
+  def how_long_urls_will_be_valid
+    1.week.to_i - 10
+  end
+
+
+  def created_at
+    request.created_at
+  end
+
+  def patron_name
+    request.patron_name
+  end
+
+  def work
+    request.work
+  end
+
+  def assets
+    work.members.order(:position).select {|x| x.is_a? Asset}
+  end
+
+  def asset_links
+    assets.map do |asset|
+      url = asset.file.url(
+        public: false,
+        expires_in: how_long_urls_will_be_valid,
+        response_content_type: asset.content_type,
+        response_content_disposition: ContentDisposition.format(
+          disposition: "inline",
+          filename: DownloadFilenameHelper.filename_for_asset(asset)
+        )
+      )
+      { text: asset.title, url: url }
+    end
+  end
+
+  def subject
+    "Science History Institute: files from #{work.title}"
+  end
+
+  def hostname
+    ScihistDigicoll::Env.lookup!(:app_url_base)
+  end
+
+  def work_url
+    @work_url ||= "#{hostname}#{Rails.application.routes.url_helpers.work_path(work.friendlier_id)}"
```

Comment:
Are you sure you have to do `Rails.application.routes.url_helpers`?  i think just `work_path` is supposed to work in ActionMailer. 

And instead of concatenating the host manually you can do:

    work_path(work.friendlier_id, host: hostname)

But even better, we can configure the default hostname globally for ActionMailer, and then just do `work_path(work.friendlier_id)`. https://guides.rubyonrails.org/action_mailer_basics.html#generating-urls-in-action-mailer-views

Reply:
No you're not supposed to be able to use *_path helpers inside of an email (hence this clumsy workaround), but you can use `work_url`. I had to add a bit of config to `config/environments/test.rb` so it would work in the tests but the default hostname is already in the other two config environments, so we're good to go.